x-pack/plugins/monitoring/server/lib/unlinked_deployments/unlinked_deployment_query_filter.js

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export const unlinkedDeploymentFilter = {
+  must: [{
+    bool: {
+      should: [
+        {
+          term: {
+            cluster_uuid: {
+              value: ''
```

Comment:
Are you suggesting we change this to
```
{
  cluster_uuid: ['']
}
```
now? or to know about this for the future?

Reply:
I intended it for the future, but maybe we should change this here too.

The `term` query/filter just needs to match a single term in the field's index to match the document. A document with `cluster_uuid: [ "xyz", "" ]` would have two terms in the `cluster_uuid` index pointing back to it: `"xyz"` and `""`. The _sender_ should _not_ send `""` to avoid this type of scenario.

However, perhaps we should _only_ look for `cluster_uuid` that do not exist, like your secondary query/filter does quite appropriately. By doing that, we'll never look for `""` and therefore we won't encourage that type of data being sent.

Reply:
I guess I'm not sure _which_ cause is happening, or more likely to happen. Is it that the `cluster_uuid` field is empty, or just missing? @ycombinator

Reply:
The query handles both scenarios right now, an empty `cluster_uuid` or one that simply does not exist. I'd love to simplify if we can to @pickypg's proposal, but I just want to make sure that will actually work with monitoring works internally with each of our stack products that can be affected (mainly Logstash and Beats).

It seems beats can/will be updated to **not** send the `cluster_uuid` (I can create an issue if you haven't) but we should verify that works with Logstash too

Reply:
> It seems beats can/will be updated to not send the cluster_uuid (I can create an issue if you haven't) but we should verify that works with Logstash too

Today its simply not possible for Beats or LS to omit the `cluster_uuid`, as that's being added by the production cluster that proxies the monitoring data to the monitoring cluster.

Going forward (with direct shipping to the monitoring cluster), the work of not sending a `cluster_uuid` for Beats is already being done in https://github.com/elastic/beats/pull/9260 (which is actually what led to this UI issue/PR if you recall 😄). For Logstash as well the work has already started (https://github.com/elastic/beats/pull/9163) but it hasn't advanced as much as Beats yet. In either case, we have issues/PRs already so we're good.

Reply:
@ycombinator 
<img src="https://media.giphy.com/media/vCKC987OpQAco/giphy.gif"/>